jax/_src/stages.py

```diff
@@ -601,26 +601,34 @@ class Lowered(Stage):
   querying properties of lowered computations across JAX's various
   lowering paths (:func:`~jax.jit`, :func:`~jax.pmap`, etc.).
   """
-  __slots__ = ["args_info", "out_tree", "_lowering", "_no_kwargs"]
+  __slots__ = ["name", "jaxpr", "args_info", "out_tree", "_lowering", "_no_kwargs"]
 
+  name: str
```

Comment:
Mark name and jaxpr as private by putting an `_` in front of them?

Reply:
How about we leave `name` as public and mark `jaxpr` as private.

Reply:
Sounds good.

Reply:
Actually maybe name should also be private for now? Making it public is easy if we want to in the future. It's not clear to me that the Lowering name should be the same as the function's name. WDYT?

Reply:
The intent is to have here the function name. Would you feel more comfortable if we called it `fun_name`?

Reply:
Yeah fun_name sounds much better! But that can also be private?

What's the reason to make it public? Do you expect users to use it?

Reply:
It's possible that `args_info` and `out_tree` don't need to be public. The latter is at least seems useful prior to us starting to pull `make_jaxpr` functionality into AOT (see `jit(f).specialize` now). It's always easy to start private.